pkg/oc/cli/cmd/importer/appjson.go

```diff
@@ -56,67 +59,68 @@ var (
-// as a template.
-func NewCmdAppJSON(fullName string, f kcmdutil.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
-	options := &AppJSONOptions{
+func NewAppJSONOptions(streams genericclioptions.IOStreams) *AppJSONOptions {
+	return &AppJSONOptions{
 		Action: configcmd.BulkAction{
-			Out:    out,
-			ErrOut: errout,
+			Out:    streams.Out,
+			ErrOut: streams.ErrOut,
 		},
-		In:        in,
-		Generator: AppJSONV1GeneratorName,
+		IOStreams:  streams,
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(ocscheme.PrintingInternalScheme),
+		Generator:  AppJSONV1GeneratorName,
 	}
+}
+
+// NewCmdAppJSON imports an app.json file (schema described here: https://devcenter.heroku.com/articles/app-json-schema)
+// as a template.
+func NewCmdAppJSON(fullName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewAppJSONOptions(streams)
 	cmd := &cobra.Command{
 		Use:     "app.json -f APPJSON",
 		Short:   "Import an app.json definition into OpenShift (experimental)",
 		Long:    appJSONLong,
 		Example: fmt.Sprintf(appJSONExample, fullName),
 		Run: func(cmd *cobra.Command, args []string) {
-			kcmdutil.CheckErr(options.Complete(f, cmd, args))
-			kcmdutil.CheckErr(options.Validate())
-			if err := options.Run(); err != nil {
-				// TODO: move me to kcmdutil
-				if err == kcmdutil.ErrExit {
-					os.Exit(1)
-				}
-				kcmdutil.CheckErr(err)
-			}
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run())
 		},
 	}
 	usage := "Filename, directory, or URL to app.json file to use"
-	kcmdutil.AddJsonFilenameFlag(cmd.Flags(), &options.Filenames, usage)
+	kcmdutil.AddJsonFilenameFlag(cmd.Flags(), &o.Filenames, usage)
 	cmd.MarkFlagRequired("filename")
-
-	cmd.Flags().StringVar(&options.BaseImage, "image", options.BaseImage, "An optional image to use as your base Docker build (must have ONBUILD directives)")
-	cmd.Flags().String("generator", options.Generator, "The name of the generator strategy to use - specify this value to for backwards compatibility.")
-	cmd.Flags().StringVar(&options.AsTemplate, "as-template", "", "If set, generate a template with the provided name")
-
-	options.Action.BindForOutput(cmd.Flags())
-	cmd.Flags().String("output-version", "", "The preferred API versions of the output objects")
+	cmd.Flags().StringVar(&o.BaseImage, "image", o.BaseImage, "An optional image to use as your base Docker build (must have ONBUILD directives)")
+	cmd.Flags().StringVar(&o.Generator, "generator", o.Generator, "The name of the generator strategy to use - specify this value to for backwards compatibility.")
+	cmd.Flags().StringVar(&o.AsTemplate, "as-template", o.AsTemplate, "If set, generate a template with the provided name")
+	cmd.Flags().StringVar(&o.OutputVersionStr, "output-version", o.OutputVersionStr, "The preferred API versions of the output objects")
```

Comment:
followup issue to see about deprecating this

Reply:
Opened issue here: https://github.com/openshift/origin/issues/20277